### DISCUSSION TITLE
Missing password and file open dialog boxes

### DISCUSSION THREAD

**Discussion by abhijeetbhagat** [emilk/egui#268] [Ideas]
I can't seem to find these controls and probably they are missing. I would love to contribute towards implementing them if someone can guide me.

**Comment by emilk** [comment]
Hi! I just added a password field in https://github.com/emilk/egui/commit/d848b2a664a7aec65ca01924c54c8b1d00da58a0

A file open dialog box would be very nice. Are you think for web or for native? They will require very different approaches.

There is also an adjacent task of drag-and-drop of files (also for both web and native), which is probably easier (and maybe more desierable).

I've added a tracking issue for this at https://github.com/emilk/egui/issues/270

**Reply by abhijeetbhagat** [reply]
> Are you think for web or for native?
Both. I can help with the implementation of both. Would the web version be achieved with the help of an html file `input` element? Or would we have to make it part of the canvas?

**Reply by emilk** [reply]
Awesome! I don’t know the best way to approach it; I’m not great with web tech

**Reply by g-braeunlich** [reply]
Have a draft of a file dialog here: https://github.com/emilk/egui/discussions/272
However: at the current stage, it is very primitive
